### ISSUE TITLE
Custom buttons or elements in header

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#496]
Originally reported on Google Code with ID 225

```
Hi,

First of let me say, I love this plugin. I think it's amazing!

I was wondering if there was a way to add a custom button to the header
view? I would like to be able to add a button labeled: Add Event that would
open a new event dialog.

Is this possible?

Thanks.
```

Reported by `raffik` on 2009-12-04 04:17:33

Imported with **27** stars

**Comment by arshaw** [comment]
the ability to add buttons has been added in v2.4.0 of fullcalendar:

http://fullcalendar.io/docs/display/customButtons/

**Comment by arshaw** [comment]
custom buttons can be put in the header now, but we can further discuss how to put other things (arbitrary widgets and html)

thus, keeping this issue open

**Comment by Stafie** [comment]
Nice addition of the custom buttons in the calendar header. 
Also I think that the ability to specify custom html for the elements added would be great.

**Comment by espen** [comment]
Yeah would be nice @Stafie, here is what I am doing currently:
`$('#fullcalendar').find('.fc-right').append( 'html' );`

**Comment by mik356ua** [comment]
Yeah it would be great to add any html code like this or something similar:
```javascript

$('#calendar').fullCalendar({
    customElements {
        myCustomDropdown: {
            html: '<div class="dropdown">\
                     <button class="btn dropdown-toggle" type="button" data-toggle="dropdown">\
                       Dropdown  <span class="caret"></span>\
                     </button>\
                     <ul class="dropdown-menu">\
                       <li><a href="#" ng-click="runAction()">Action</a></li>\
                       <li><a href="#" ng-click="runAnotherAction()">Another action</a></li>\
                     </ul>\
                   </div>'
        },
        myCustomButton: {
            type: 'button',
            text: 'custom!',
            click: function() {
                alert('clicked the custom button!');
            }
        }
    },
    header: {
        left: 'prev,next today myCustomButton',
        center: 'title',
        right: 'myCustomDropdown month,agendaWeek,agendaDay'
    }
});
```
And thank you for your awesome plugin:)

**Comment by flyingangel** [comment]
Custom button should receive a HTML code, element or object

    $('#calendar').fullCalendar({
       customButtons: {
            myCustomButton: {
                text: '<div>some div</div>!',   //as html string
                type: 'html',
                html: object    //direct passing html element, will override text
            }
        },
    });

    object receive a DOM element, string, jQuery object, jQuery selector as parameter

**Comment by Liaohuanle** [comment]
const customButtons = {
    watchButton: {
      text: 'watch',
      type: 'button',
      click: function () {
        alert('clicked the custom button!')
      }
    },
    userButton: {
      text: 'mine',
      click: function () {
        alert('clicked the custom button!')
      }
    }
  }

  var calendar = $('#calendar').fullCalendar({
    customButtons,
    header: {
      left: 'title',
      center: 'watchButton,userButton',
      right: 'prev,next today'
    },
.......
i have try each way,but its still can not append to the view.

**Comment by bradbajuz** [comment]
What's the status on this?

I'm currently trying to figure out the best way to render the right button group as a dropdown menu with Bootstrap.

Bootstrap already as has a button dropdown component (dropdown-toggle and dropdown-menu). Wonder if there should be a way to just hook into the existing Bootstrap framework by a simple change. For example, in the existing header hash, give the ability to choose which position should be a dropdown or a group toggle. 

```
header: {
  left: 'prev,next today',
  center: 'title',
  right: 'agendaDay,agendaWeek,month',
  toggle: {dropdown: 'right'}
}

```

**Comment by frlinw** [comment]
@bradbajuz `header: false` then you can do it yourself thanks to fullcalendar methods
a quick & dirty example: https://codepen.io/frlinw/pen/opdoeO

**Comment by bradbajuz** [comment]
Great. Thank you @frlinw

**Comment by frlinw** [comment]
As the previous demo got some popularity (+3,4k), I did a demo for the v4
- v3: https://codepen.io/frlinw/pen/opdoeO
- v4: https://codepen.io/frlinw/pen/VNbRLL

**Comment by acerix** [comment]
For reference, an example to add elements to the header from `viewSkeletonRender`:

https://codepen.io/acerix/pen/jOOgQeL?editors=0010

**Comment by TejasCMehta** [comment]
> Suggestion: Custom button should receive a HTML code, element or object
> 
> ```
> $('#calendar').fullCalendar({
>    customButtons: {
>         myCustomButton: {
>             text: '<div>some div</div>!',   //as html string
>             type: 'html',
>             html: object    //direct passing html element, will override text
>         }
>     },
> });
> 
> object receive a DOM element, string, jQuery object, jQuery selector as parameter
> ```

How to do this in Angular 8 ?

**Comment by acerix** [comment]
For reference, can be done in v5 with `viewDidMount`:

https://codepen.io/acerix/pen/YzEdxGx?editors=001

**Comment by jrock2004** [comment]
> For reference, can be done in v5 with `viewDidMount`:
> 
> https://codepen.io/acerix/pen/YzEdxGx?editors=001

I understand that this might work if you are building in straight JS but something like react this would not work if you want to mount a component vs straight HTML. I believe this should be a feature request

**Comment by venubabu1996** [comment]
how can i add  dropdown  like month week and day view instead of separate button in ts

**Comment by PrinceAnyichie** [comment]
Please is there a demo showing how to add custom element particularly a dropdown or select element in Angular?

**Comment by BrunoElo** [comment]
Please is there a demo showing how to add custom element particularly a dropdown or select element in Angular?

**Comment by venubabu1996** [comment]
![Screenshot 2022-06-08 132554](https://user-images.githubusercontent.com/51917091/172563181-bf8fee6c-3297-42c5-9686-c52bf5e8de72.png)

**Comment by venubabu1996** [comment]
you can refer month view mockup

**Comment by venubabu1996** [comment]
> 

any solution

**Comment by moabbar** [comment]
Any updates on how to add a dropdown filter to the header toolbar?

**Comment by jrock2004** [comment]
> Is there any progress on adding a dropdown filter to the header toolbar?
> I've created a new issue for this feature. If you find it helpful and necessary, please show your support by giving it a thumbs-up 👍 emoji on the following link: #7356

Make your own header is your only option.

**Comment by arshaw** [comment]
This will be implemented for v7. More info: https://github.com/fullcalendar/fullcalendar/issues/7907

**Comment by Latysovictomas** [comment]
Any chance to add Angular content injection support to replace full-calendar toolbar buttons with our own? We use custom-designed buttons and want to use our own directives for styling and custom tooltips designs. (related: https://github.com/fullcalendar/fullcalendar/issues/5779)
